hyperspy/io_plugins/semper_unf.py

```diff
@@ -0,0 +1,516 @@
+                assert struct.unpack('>h', ''.join([chr(x) for x in label[10:12]]))[0] == nlay
+                iccoln = struct.unpack('>h', ''.join([chr(x) for x in label[12:14]]))[0]
+                icrown = struct.unpack('>h', ''.join([chr(x) for x in label[14:16]]))[0]
+                iclayn = struct.unpack('>h', ''.join([chr(x) for x in label[16:18]]))[0]
+                assert label[18] == iclass
+                assert label[19] == iform
+                iwp = label[20]
+                date = '{}-{}-{} {}:{}:{}'.format(label[21]+1900, *label[22:27])
+                # No test for ncrang, range is extracted from data itself (also prone to errors)!
+                ipltyp = label[55]  # position list type
+                real_coords = label[62]
+                dz, dy, dx, z0, y0, x0 = [1., 1., 1., 0., 0., 0.]
+                if real_coords:
+                    dz = struct.unpack('<f', ''.join([chr(x) for x in label[75:79]]))[0]
+                    z0 = struct.unpack('<f', ''.join([chr(x) for x in label[79:83]]))[0]
+                    dy = struct.unpack('<f', ''.join([chr(x) for x in label[83:87]]))[0]
+                    y0 = struct.unpack('<f', ''.join([chr(x) for x in label[87:91]]))[0]
+                    dx = struct.unpack('<f', ''.join([chr(x) for x in label[91:95]]))[0]
+                    x0 = struct.unpack('<f', ''.join([chr(x) for x in label[95:99]]))[0]
+                assert ''.join([str(unichr(l)) for l in label[100:100+ntitle]]) == title
+                ux = ''.join([chr(l) for l in label[244:248]]).replace('\x00', '')
+                uy = ''.join([chr(l) for l in label[248:252]]).replace('\x00', '')
+                uz = ''.join([chr(l) for l in label[252:256]]).replace('\x00', '')
+            assert np.frombuffer(f.read(4), dtype=np.int32)[0] == rec_length
+            # Read picture data:
+            data = np.empty((nlay, nrow, ncol), dtype=data_format)
+            for k in range(nlay):
+                for j in range(nrow):
+                    rec_length = np.frombuffer(f.read(4), dtype=np.int32)[0]  # length of row
+                    row = np.frombuffer(f.read(rec_length), dtype=data_format)
+                    data[k, j, :] = row
+                    assert np.frombuffer(f.read(4), dtype=np.int32)[0] == rec_length
+        arg_dict = {}
+        arg_dict['data'] = data
+        arg_dict['title'] = title
+        arg_dict['offsets'] = (x0, y0, z0)
+        arg_dict['scales'] = (dx, dy, dz)
+        arg_dict['units'] = (ux, uy, uz)
+        arg_dict['date'] = date
+        arg_dict['ICLASS'] = iclass
+        arg_dict['IFORM'] = iform
+        arg_dict['IVERSN'] = iversn
+        arg_dict['ILABEL'] = ilabel
+        arg_dict['IFORMAT'] = iformat
+        arg_dict['IWP'] = iwp
+        arg_dict['IPLTYP'] = ipltyp
+        arg_dict['ICCOLN'] = iccoln
+        arg_dict['ICROWN'] = icrown
+        arg_dict['ICLAYN'] = iclayn

```

Comment:
Since it's just assignment, it's more efficient to create the dictionary already full, so that it reads
```python
arg_dict = { 'data': data,
             'title': title,
             'offsets': (x0, y0, z0),
             'scales': (dx, dy, dz),
             'units': (ux, uy, uz),
             'date': date,
             'ICLASS': iclass,
             'IFORM': iform,
             'IVERSN': iversn,
             'ILABEL': ilabel,
             'IFORMAT': iformat,
             'IWP': iwp,
             'IPLTYP': ipltyp,
             'ICCOLN': iccoln,
             'ICROWN': icrown, 
             'ICLAYN': iclayn
            }
```
instead. do check for spelling / similar errors if copy-pasting